docs/remote_inference_blueprints/aleph_alpha_connector_luminous_base_embedding_blueprint.md

```diff
@@ -0,0 +1,143 @@
+      "headers": {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "Authorization": "Bearer ${credential.AlephAlpha_API_Token}"
+      },
+      "request_body": "{ \"model\": \"luminous-base\", \"prompt\": \"${parameters.input}\", \"representation\": \"${parameters.representation}\", \"normalize\": ${parameters.normalize}}",
+      "pre_process_function": "\n    StringBuilder builder = new StringBuilder();\n    builder.append(\"\\\"\");\n    String first = params.text_docs[0];\n    builder.append(first);\n    builder.append(\"\\\"\");\n    def parameters = \"{\" +\"\\\"input\\\":\" + builder + \"}\";\n    return  \"{\" +\"\\\"parameters\\\":\" + parameters + \"}\";",
+      "post_process_function": "\n      def name = \"embedding\";\n      def dataType = \"FLOAT32\";\n      if (params.embedding == null || params.embedding.length == 0) {\n        return params.message;\n      }\n      def shape = [params.embedding.length];\n      def json = \"{\" +\n                 \"\\\"name\\\":\\\"\" + name + \"\\\",\" +\n                 \"\\\"data_type\\\":\\\"\" + dataType + \"\\\",\" +\n                 \"\\\"shape\\\":\" + shape + \",\" +\n                 \"\\\"data\\\":\" + params.embedding +\n                 \"}\";\n      return json;\n    "
+    }
+  ]
+}
+```
+
+Sample response:
+```json
+{
+  "connector_id": "bRa3QI0BWgGoN0Ye9K2u"
+}
+```
+
+## 3. Create model group:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "remote_model_group",
+    "description": "This is an example description"
+}
+```
+
+Sample response:
+```json
+{
+  "model_group_id": "XRbIP40BWgGoN0YeZq1V",
+  "status": "CREATED"
+}
+```
+
+## 4. Register model to model group & deploy model:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "Luminous-base embedding model",
+  "function_name": "remote",
+  "model_group_id": "XRbIP40BWgGoN0YeZq1V",
+  "description": "embedding model, representation: document",
+  "connector_id": "bRa3QI0BWgGoN0Ye9K2u"
+}
+```
+
+Sample response:
+```json
+{
+  "task_id": "r6R9PIsBQRofe4CSlUoG",
+  "status": "CREATED"
+}
+```
+Get model id from task
+```json
+GET /_plugins/_ml/tasks/r6R9PIsBQRofe4CSlUoG
+```
+Deploy model, in this demo the model id is `sKR9PIsBQRofe4CSlUov`
+```json
+POST /_plugins/_ml/models/sKR9PIsBQRofe4CSlUov/_deploy
+```
+
+## 5. Test model inference
+
+```json
+POST /_plugins/_ml/models/sKR9PIsBQRofe4CSlUov/_predict
+{
+  "parameters": {
+    "input": "Test string with German characters: Moët Hennessy"
+  }
+}
+```
+
+Sample response:
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "embedding",
+          "data_type": "FLOAT32",
+          "shape": [
+            5120
+          ],
+          "data": [
+            -0.012756348,
+            0.001159668,
+            0.0025634766,
+            ...
+          ]
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
```

Comment:
Can you also try if this can work ? This is to make sure neural-search plugin can work with ml-commons.

```
POST /_plugins/_ml/_predict/text_embedding/sKR9PIsBQRofe4CSlUov
{
  "text_docs":[ "How are you", "hello world"],
  "return_number": true,
  "target_response": ["sentence_embedding"]
}
```
The expected response format
```
{
  "inference_results" : [
    {
      "output" : [
        {
          "name" : "sentence_embedding",
          "data_type" : "FLOAT32",
          "shape" : [
            384
          ],
          "data" : [
            -0.023315024,
            0.08975691,
            0.078479774,
            ...
          ]
        }
      ]
    }
  ]
}
```

Reply:
Sure thing. I just checked, the result is this:
```json
{
    "inference_results": [
        {
            "output": [
                {
                    "name": "sentence_embedding",
                    "data_type": "FLOAT32",
                    "shape": [
                        5120
                    ],
                    "data": [
                        -0.0038909912,
                        0.016845703,
                        -0.014770508,
						...
                    ]
                }
            ],
            "status_code": 200
        },
        {
            "output": [
                {
                    "name": "sentence_embedding",
                    "data_type": "FLOAT32",
                    "shape": [
                        5120
                    ],
                    "data": [
                        0.0039978027,
                        0.0030670166,
                        2.1457672E-4,
						...
                   ]
                }
            ],
            "status_code": 200
        }
    ]
}
```